VideoApp/VideoApp/Stores/MainParticipant/MainParticipantStore.swift

```diff
@@ -17,7 +17,26 @@
 import Foundation
 
 class MainParticipantStore {
-    private(set) var mainParticipant: Participant
+    private(set) var mainParticipant: Participant {
+        didSet {
+            videoTrack = mainParticipant.mainVideoTrack
+
+            if mainParticipant.isPinned || videoTrack === mainParticipant.screenTrack || !mainParticipant.isDominantSpeaker {
+                videoTrack?.priority = .high
+            } else {
+                videoTrack?.priority = nil
+            }
+
+            notificationCenter.post(name: .mainParticipantStoreUpdate, object: self)
+        }
+    }
+    private(set) var videoTrack: VideoTrack? {
+        didSet {
+            guard oldValue !== videoTrack else { return }
+
+            oldValue?.priority = nil
+        }
+    }
```

Comment:
This file got quite a lot simpler even though it's doing more work because:

1. Using `didSet` is really nice for comparing new and old values. This kind of logic was in a separate function and I had to keep track of my own new and old state.
1. The update handling was really overly optimized and is not just more code but it was getting a little hard to follow. The logic to calculate the main participant is O(n) and the UI updates are tiny and it is only updating one view so really we can just process any update the same way.